src/main/java/hillel/spring/doctors/DoctorsRestController.java

```diff
@@ -1,46 +1,59 @@
 package hillel.spring.doctors;
 
 
+import hillel.spring.doctors.dto.DoctorInputDto;
+import hillel.spring.doctors.dto.DoctorsDtoMapper;
 import hillel.spring.doctors.exceptions.DoctorIsNotFoundException;
-import hillel.spring.doctors.exceptions.IdIsNotEqualToUpdateDoctorException;
-import hillel.spring.doctors.exceptions.IdIsPresentToCreateException;
 import hillel.spring.doctors.model.Doctor;
 import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.val;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
+import org.springframework.web.util.UriComponentsBuilder;
 
 import java.util.List;
 import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
 
 @RestController
 @AllArgsConstructor
 public class DoctorsRestController {
 
     private final DoctorRestService doctorRestService;
+    @Autowired
+    private final DoctorsDtoMapper dtoConverter;
+
+    private final UriComponentsBuilder uriBuilder = UriComponentsBuilder.newInstance()
+            .scheme("http")
+            .host("localhost")
+            .path("/doctors/{id}");
 
     @GetMapping("/doctors")
-    public List<Doctor> findAllDoctors() {
+    public List<Doctor> findAllDoctors(@RequestParam(value = "letter", required = false) Optional<String> letter,
+                                       @RequestParam(value = "specialization", required = false) Optional<String> specialization) {
```

Comment:
Rince parameters are of type `Optional`  it means they are not required by default. So `required=false`  is redundant